pkg/apis/machine/v1beta1/machine_webhook_test.go

```diff
@@ -229,3 +230,735 @@ func TestDefaultAWSProviderSpec(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateAzureProviderSpec(t *testing.T) {
+
+	testCases := []struct {
+		testCase      string
+		providerSpec  *azure.AzureMachineProviderSpec
+		expectedError string
+		expectedOk    bool
+	}{
+		{
+			testCase: "with no location it fails",
+			providerSpec: &azure.AzureMachineProviderSpec{
```

Comment:
Most of test cases are using valid provider spec, but then altering a single field. Could we just patch full spec with changes we need to test, to improve readability?